qualang_tools/optimal_weights/configuration.py

```diff
@@ -0,0 +1,271 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+#############################
+# simulation helpers #
+#############################
+
+
+def simulate_pulse(IF_freq, chi, k, Ts, Td, power):
+    I = [0]
+    Q = [0]
+
+    for t in range(Ts):
+        I.append(I[-1]+(power/2 - k*I[-1]+Q[-1]*chi))
+        Q.append(Q[-1]+(power/2 - k*Q[-1]-I[-1]*chi))
+
+    for t in range(Td - 1):
+        I.append(I[-1]+(-k*I[-1]+Q[-1]*chi))
+        Q.append(Q[-1]+(-k*Q[-1]-I[-1]*chi))
+
+    I = np.array(I)
+    Q = np.array(Q)
+    t = np.arange(len(I))
+
+    S = I * np.cos(2 * np.pi * IF_freq * t * 1e-9) + Q * np.sin(2 * np.pi * IF_freq * t * 1e-9)
+
+    return t, I, Q, S
+
+
+resonators_lo = 7.1e9  # High Band Pass
+WG1_lo = resonators_lo
+WG2_lo = resonators_lo
+q1a_res_IF = 40e6
+q2a_res_IF = 50e6
+
+
+readout_len = 480
+IF_freq = q1a_res_IF
+Ts = readout_len-200
+Td = 200
+power = 0.2
+k = 0.04
+chi = 0.023
+[tg_, Ig_, Qg_, Sg_] = simulate_pulse(IF_freq, 1 * chi, k, Ts, Td, power)
+[te_, Ie_, Qe_, Se_] = simulate_pulse(IF_freq, 3 * chi, k, Ts, Td, power)
+[tf_, If_, Qf_, Sf_] = simulate_pulse(IF_freq, 5 * chi, k, Ts, Td, power)
+
+# plt.figure()
+# plt.plot(Ig_, Qg_)
+# plt.plot(Ie_, Qe_)
+# plt.plot(If_, Qf_)
+#
+# plt.figure()
+# plt.plot(Ig_)
+# plt.plot(Qg_)
+# plt.plot(Ie_)
+# plt.plot(Qe_)
+# plt.plot(If_)
+# plt.plot(Qf_)
```

Comment:
we shouldn't have commented code here

Reply:
Maybe, it makes too many figures and may not be relevant though, but why note -- just need to make it have a nice title and labels

Reply:
That's OK.
I'm saying we need to not have commented code: so if it is not needed, delete it!